test/test_testmon.py

```diff
@@ -192,6 +192,25 @@ def test_add():
         assert tuple(res) == (1, 0, 0), res
         sys.modules.pop('test_a', None)
 
+    def test_tlf(self, testdir):
+        tf = testdir.makepyfile(test_a="""
+            def test_add():
+                1/0
+        """, )
+        testdir.inline_run("--testmon", "-v")
+        sys.modules.pop('test_a', None)
+
+        result = testdir.runpytest("--testmon", "-v")
+        result.stdout.fnmatch_lines([
+            "*1 failed, 1 deselected*",
+        ])
+        tf.setmtime(1424880936)
+        result = testdir.runpytest("--testmon", "-v", "--tlf")
+        result.stdout.fnmatch_lines([
+            "*1 failed in*",
+        ])
```

Comment:
If you wanted a more involved test case with external factors I would recommend somthing like:
  ```python
    os.environ['TLF']='ON'
    tf = testdir.makepyfile(test_a="""
            import os
            def test_add():
                if os.environ.get('TLF', None):
                    raise Exception('TLFON')
        """, )
    testdir.inline_run("--testmon", "-v")

```